### PR TITLE
Change usePythonVersion function to choose pypy version

### PR DESCRIPTION
**Pull request by SvetlanaMaliugina**
**Task name**: UsePythonVersionV0

**Description**:  
- Give user opportunity to choose version of PyPy on input ```versionSpec``` of UsePythonVersionV0 task. For that changed usePythonVersion function according to contributor's PR : Update UsePythonVersion to allow versions other than (pypy2 or pypy3.6) #15553.
- Added warning message for specifying an exact version of PyPy. 
- Updated patch version.

**Documentation changes required:** Y

**Added unit tests:** N

**Attached related issue:** Support PyPy 3.7 installation by UsePythonVersion@0 tasks. #15232

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

### CODE REVIEW

File: Tasks/UsePythonVersionV0/task.json [hunk 1]
```diff
@@ -1,123 +1,124 @@
...
-    "name": "UsePythonVersion",
-    "friendlyName": "Use Python version",
-    "description": "Use the specified version of Python from the tool cache, optionally adding it to the PATH",
-    "helpUrl": "https://docs.microsoft.com/azure/devops/pipelines/tasks/tool/use-python-version",
-    "helpMarkDown": "[Learn more about this task](https://go.microsoft.com/fwlink/?linkid=871498) or [see the Python documentation](https://www.python.org/doc/)",
-    "category": "Tool",
-    "runsOn": [
-        "Agent",
-        "DeploymentGroup"
+  "id": "33C63B11-352B-45A2-BA1B-54CB568A29CA",
```
Comment (anatolybolshakov) [inline Tasks/UsePythonVersionV0/task.json:2]: Are all of these changes relevant?
Could you please remove redundant indent changes?

File: Tasks/UsePythonVersionV0/usepythonversion.ts [hunk 2]
```diff
@@ -157,13 +167,12 @@ async function useCpythonVersion(parameters: Readonly<TaskParameters>, platform:
...
 export async function usePythonVersion(parameters: Readonly<TaskParameters>, platform: Platform): Promise<void> {
-    switch (parameters.versionSpec.toUpperCase()) {
-        case 'PYPY2':
-            return usePyPy('2', parameters, platform);
-        case 'PYPY3':
-            // keep pypy3 pointing to 3.9 for backward compatibility
-            return usePyPy('3.9', parameters, platform);
-        default:
-            return await useCpythonVersion(parameters, platform);
+    let fullVersionSpec: string = parameters.versionSpec.toUpperCase();
```
Comment (KonstantinTyukalov) [inline Tasks/UsePythonVersionV0/usepythonversion.ts:170]: As I see no mutation behavior. Can we use const instead?

**Review by KonstantinTyukalov — APPROVED** [review]
LGTM in general, thanks! Please check comments

**Review by kirill-ivlev — APPROVED** [review]

File: Tasks/UsePythonVersionV0/usepythonversion.ts [hunk 3]
```diff
@@ -157,13 +164,12 @@ async function useCpythonVersion(parameters: Readonly<TaskParameters>, platform:
...
-        case 'PYPY3':
-            // keep pypy3 pointing to 3.9 for backward compatibility
-            return usePyPy('3.9', parameters, platform);
-        default:
-            return await useCpythonVersion(parameters, platform);
+    const fullVersionSpec: string = parameters.versionSpec.toUpperCase();
+
+    if (fullVersionSpec.startsWith("PYPY")) {
+        //Trim off the beginning PYPY and look for it by version
+        return  usePyPy(fullVersionSpec.substring(4), parameters, platform);
```
Comment (max-zaytsev) [inline Tasks/UsePythonVersionV0/usepythonversion.ts:171]: Extra spaces

File: Tasks/UsePythonVersionV0/usepythonversion.ts [hunk 4]
```diff
@@ -49,10 +49,17 @@ function pypyNotFoundError(versionSpec: '2' | '3.9') {
 // For example, PyPy 7.0 contains Python 2.7, 3.5, and 3.6-alpha.
 // We only care about the Python version, so we don't use the PyPy version for the tool cache.
 
-function usePyPy(versionSpec: '2' | '3.9', parameters: TaskParameters, platform: Platform): void {
+function usePyPy(versionSpec: string, parameters: TaskParameters, platform: Platform): void {
     const findPyPy = tool.findLocalTool.bind(undefined, 'PyPy', versionSpec);
     let installDir: string | null = findPyPy(parameters.architecture);
 
+    const desugaredVersionSpec = desugarDevVersion(versionSpec);
```
Comment (max-zaytsev) [inline Tasks/UsePythonVersionV0/usepythonversion.ts:56]: Types

**Review by max-zaytsev — APPROVED** [review]

**Review by wojgasior — APPROVED** [review]

**Pull request merged**